src/main/java/seedu/address/model/person/Student.java

```diff
@@ -67,10 +77,19 @@ public ClassTiming getClassTiming() {
      */
     @Override
     public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Student)) {
+            return false;
+        }
+
+        Student o = (Student) other;
         return super.equals(other)
-                && other instanceof Student
-                && ((Student) other).getClassTiming().equals(getClassTiming())
-                && ((Student) other).getTags().equals(getTags());
+                && o.rate.equals(getRate())
+                && o.classTiming.equals(getClassTiming())
+                && o.getTags().equals(getTags());
```

Comment:
Good idea to take out (Student) other as a variable, so we do not have to keep type casting it.